modules/FvwmMFL/FvwmMFL.c

```diff
@@ -623,23 +628,52 @@ static void
 fvwm_read(int efd, short ev, void *data)
 {
 	FvwmPacket	*packet;
+	char *sock_path;
+
+	sock_path = set_socket_path();
 
 	if ((packet = ReadFvwmPacket(efd)) == NULL) {
 		if (debug)
 			fprintf(stderr, "Couldn't read from FVWM - exiting.\n");
-		unlink(SOCK);
+		unlink(sock_path);
 		exit(0);
 	}
 
 	broadcast_to_client(packet);
 }
 
+static char *set_socket_path()
```

Comment:
```suggestion
static const char *
set_socket_path(void)
```

Reply:
Hi Thomas,

As you pointed, it is a full filesystem address, so I have renamed sock_path into sock_pathname, made freeing of variable where you pointed it out, *sock_pathname is now global and at the top with other similar variables, but I have one problem:

I have changed everything you pointed out, but if I make "const" socket_pathname and put const function set_socket_pathname prototype and body, code complains but xasprintf complains (see below). I don't know how to resolve this except of not making this "const".

  CC       FvwmMFL.o
FvwmMFL.c: In function ‘set_socket_pathname’:
FvwmMFL.c:662:16: warning: passing argument 1 of ‘xasprintf’ from incompatible pointer type [-Wincompatible-pointer-types]
  662 |      xasprintf(&sock_pathname, "%s/%s", getenv("FVWM_USERDIR"), unrolled_path);
      |                ^~~~~~~~~~~~~~
      |                |
      |                const char **
In file included from ../../libs/fvwmlib.h:13,
                 from ../../libs/Flocale.h:14,
                 from ../../fvwm/fvwm.h:43,
                 from FvwmMFL.c:11:
../../libs/safemalloc.h:7:16: note: expected ‘char **’ but argument is of type ‘const char **’
    7 | int  xasprintf(char **, const char *, ...);
      |                ^~~~~~~
FvwmMFL.c: In function ‘main’:
FvwmMFL.c:694:7: warning: passing argument 1 of ‘free’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  694 |  free(sock_pathname);
      |       ^~~~~~~~~~~~~
In file included from ../../config.h:474,
                 from FvwmMFL.c:9:
/usr/include/stdlib.h:565:25: note: expected ‘void *’ but argument is of type ‘const char *’
  565 | extern void free (void *__ptr) __THROW;
      |                   ~~~~~~^~~~~
  CCLD     FvwmMFL